package.md

```diff
@@ -0,0 +1,42 @@
+# package.json notes
+
+This package is bundled as both a commonjs and es6-compatible NPM bundle. The
+factor that enables this dual packaging is the "exports" key in the package.json
+file:
+
+## exports
+
+- `require`: When this package is in a commonjs environment (e.g. default nodejs) 
+  the ./index.js file will be the thing that is 'require'd.
+- `import`: When this package is in an es6 environment (e.g. bundler, modern nodejs,
+  modern browser) the './esm/wrapper.js will be the thing 'import'ed.
+
+## dependencies
+
+
+  
+- `node-fetch`: When the http-client protocol is used, node-fetch supplies 'fetch'
+  for a nodejs client
+- `node-web-streams`: Although node-fetch is mostly isomorphic, its internal stream
+  is not the same as a web stream reader; we need it to have the same API.
+- `spdy`: This gives us http2.0 connection multiplexing with a 'natural http module
+  interface'. (http1.1 provides a max of 6 open conns)
+
+## Development Notes
+
+For code that is intended to run in all environments (e.g. browser, node) and
+potentially pass through a bundler step, the following guidelines are helpful:
+
+- Use single-value module.exports in files, and named exports in wrappers.
+- If using globals, it's also important to use module.exports; for example: 
+ 
+```
+function braid_fetch(...) { ... }
+
+if (typeof module !== 'undefined' && module.exports) {
+    module.exports = braid_fetch
+}
+```
+
+For a complete list of reasons for the madness, and to learn more about the method
+we've used to build this package, see https://redfin.engineering/node-modules-at-war-why-commonjs-and-es-modules-cant-get-along-9617135eeca1
```

Comment:
Ah, this redfin link is a good resource, and this package.md is prescient— this javascript require/import world is confusing enough for most of us that I am already glad to have the documentation.